app/Services/Entity/RemindableService.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace App\Services\Entity;
+
+use App\Models\Calendar;
+use App\Models\Entity;
+use App\Models\EntityEventType;
+use App\Models\Post;
+use App\Models\Reminder;
+use Exception;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Http\Request;
+
+class RemindableService
+{
+    protected Request $request;
+
+    public function __construct(Request $request)
+    {
+        $this->request = $request;
+    }
+
+    public function processSaved(Post|Entity $model)
+    {
+
+        // The user is editing an entity with a calendar, but doesn't have the permission to see
+        // the calendar? We skip any work.
+        if (request()->has('calendar_skip')) {
```

Comment:
Don't call `request()` in a service but `$this->request` (and pass it in the observer)